src/heating.hpp

```diff
@@ -461,11 +485,29 @@ class Heating {
     TipSensorStatus getTipSensorStatus() const {
         return _tip_sensor_status;
     }
+ 
+    /** Getter tip type
+    indicate whether RTM or RTU tip is detected
+    
+    Return:
+        state from enum TipType
+    */
+    TipType getTipType() const {
+        return _tip_type;
+    }
 
     /** Initialize module
     */
     void init() {
-        _pid.set_constants(PID_K_PROPORTIONAL, PID_K_INTEGRAL, PID_K_DERIVATE, PERIOD_TIME_MS, HEATING_POWER_MAX_MW);
+        _max_power_mw = (40 + (110 * _settings.get_high_power())) * 1000;  // mW
+     /* Below will use tip resistance to set max power, but need to figure out how to calculate tip resistance before init, or how to change max power after init   
```

Comment:
Yes, I assume that RTU will work better with different PIDs. I am currently traveling for work, and my RTU tip arrived at the house the day after I left, so it will be about two weeks before I can plug it in to test how it works. I also am not sure I have the resources to test the temperature and create a curve to see how much damping or overshoot needs to be adjusted. But if needed I think I may be able to find a friend that can. No promises, though.

BUT, even if it did require different PIDs, I envision that we would have separate constants for both, and then in the IF statement that decides which max power to allow, it would also select the current PIDs. In my mind, finding better PIDs and the maximum power logic are two separate problems to be solved that don't need to be solved at the same time (unless I test the RTU and find that it the current PIDs don't work at all).

Reply:
Yes, btw, for tuning PID constants I have small python tool, which show diagrams of heating and so... (is not currently published)

Reply:
> > SUPPLY_VOLTAGE_MAX_MV = 18000; // mV
> 
> Yes, you are right, I did not implemented this, I forgotten.
> but voltage limits is important to implement, my plan is to disable ON if the voltage limit is over maximum allowed.

Thinking about it now, this is more complicated than I thought. I'm looking for a good place that _supply_voltage_mv_idle could be checked to force a STOP state, but then of course it will also need some notification to the user. So I will chew on this some more and find a good way to implement it.